pkg/apis/config/v1alpha1/types.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AWSCloudConfig stores the provided configuration for a cloud provider.
+type AWSCloudConfig struct {
+	metav1.TypeMeta `json:",inline"`
+
+	Config AWSCloudConfigSpec `json:"spec"`
+}
+
+// AWSCloudConfigSpec is the custom spec for an AWSCloudConfigresource
+type AWSCloudConfigSpec struct {
```

Comment:
Remove Spec from this name

Reply:
Or just name this `CloudConfig`, I think the "AWS" part is redundant since it is under AWSCloudConfig already

Reply:
Or just `Config` would be fine I think